paginated_reader.go

```diff
@@ -0,0 +1,81 @@
+package timber
+
+import (
+	"context"
+	"io"
+	"net/http"
+
+	"github.com/peterhellberg/link"
+	"github.com/pkg/errors"
+)
+
+// paginatedReadCloser implements the io.ReadCloser interface, wrapping the
+// io.ReadCloser from an HTTP response. The reader will first exhaust the
```

Comment:
I feel like it might be clearer to this said "wrapping the body from an HTTP response".